### PR TITLE
feat: Project-environment type configs  relationship

### PR DESCRIPTION
**Pull request by FernandoAranda**
Issue #, if available:
GALI-2025
Description of changes:
-Add associate API for project-etc
-Add disassociate API for project-etc
-Add list API for project-etc relationship

Checklist:

* [x] Have you successfully deployed to an AWS account with your changes?
* [x] Have you written new tests for your core changes, as applicable?

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.

### CODE REVIEW

File: solutions/swb-reference/integration-tests/tests/isolated/projectEnvTypeConfigs/list.test.ts [hunk 1]
```diff
@@ -0,0 +1,115 @@
...
+  });
+
+  test('list projectEnvTypeConfigs excecutes successfully', async () => {
+    const { data: response } = await adminSession.resources.projects
+      .project(projectId)
+      .environmentTypes()
+      .environmentType(envTypeId)
+      .configurations()
+      .get({});
+    expect(Array.isArray(response.data)).toBe(true);
```
Comment (aws-tyler) [inline solutions/swb-reference/integration-tests/tests/isolated/projectEnvTypeConfigs/list.test.ts:37]: is there any validation to do on the data?

File: solutions/swb-reference/src/backendAPI.ts [hunk 2]
```diff
@@ -25,14 +25,23 @@ import { Express } from 'express';
...
 const accountService: AccountService = new AccountService(aws.helpers.ddb);
-
+const envTypeService: EnvironmentTypeService = new EnvironmentTypeService(aws.helpers.ddb);
+const envTypeConfigService: EnvironmentTypeConfigService = new EnvironmentTypeConfigService(
+  envTypeService,
+  aws.helpers.ddb
+);
+const metadataService: MetadataService = new MetadataService(aws.helpers.ddb);
+const projectService: ProjectService = new ProjectService({
+  TABLE_NAME: process.env.STACK_NAME!
```
Comment (aws-tyler) [inline solutions/swb-reference/src/backendAPI.ts:43]: don't need to do this now but we should update ProjectService to take a dynamo service

**Review by aws-tyler — DISMISSED** [review]

File: solutions/swb-reference/integration-tests/tests/isolated/projectEnvTypeConfigs/list.test.ts (hunk 1 above)
Comment (FernandoAranda) [reply solutions/swb-reference/integration-tests/tests/isolated/projectEnvTypeConfigs/list.test.ts:37]: as this is isolated we cannot be certain there will be data, maybe we can add a list on the multiStep and we would be able to validate data

File: solutions/swb-reference/src/backendAPI.ts (hunk 2 above)
Comment (FernandoAranda) [reply solutions/swb-reference/src/backendAPI.ts:43]: I think there is a feat branch for project, so thats probably handled there

File: solutions/swb-reference/integration-tests/tests/isolated/projectEnvTypeConfigs/list.test.ts (hunk 1 above)
Comment (FernandoAranda) [reply solutions/swb-reference/integration-tests/tests/isolated/projectEnvTypeConfigs/list.test.ts:37]: Added data check for list on multiStep integ test

**Review by aws-tyler — APPROVED** [review]

**Pull request merged**